<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1350</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_12">
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Composer</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="1" column="0">
           <widget class="QGroupBox" name="groupBox_2">
            <property name="minimumSize">
             <size>
              <width>900</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>MS Serif</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="title">
             <string>Generated Signal</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <property name="topMargin">
              <number>16</number>
             </property>
             <item row="0" column="0">
              <widget class="PlotWidget" name="widget2" native="true">
               <property name="minimumSize">
                <size>
                 <width>600</width>
                 <height>100</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox">
            <property name="minimumSize">
             <size>
              <width>900</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>MS Serif</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="title">
             <string>Signal Composer</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <property name="topMargin">
              <number>17</number>
             </property>
             <item row="0" column="0">
              <widget class="PlotWidget" name="widget1" native="true">
               <property name="minimumSize">
                <size>
                 <width>700</width>
                 <height>100</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <widget class="QWidget" name="widget" native="true">
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>200</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>900</height>
           </size>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="7" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>10</width>
               <height>10</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="6" column="0">
            <widget class="QPushButton" name="confirm_button">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Send to Viewer</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QComboBox" name="signal_combobox">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="editable">
                <bool>false</bool>
               </property>
               <property name="placeholderText" stdset="0">
                <string>Select signal component</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="remove_button">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Remove</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="0">
            <layout class="QFormLayout" name="formLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>11</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>22</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="font">
                <font>
                 <family>Modern No. 20</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Frequency</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="frequency">
               <property name="minimumSize">
                <size>
                 <width>2</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="4" column="0">
            <widget class="QPushButton" name="add_button">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <layout class="QFormLayout" name="formLayout_2">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>11</number>
             </property>
             <property name="bottomMargin">
              <number>22</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_2">
               <property name="font">
                <font>
                 <family>Modern No. 20</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="lineWidth">
                <number>0</number>
               </property>
               <property name="text">
                <string>Magnitude</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="magnitude">
               <property name="minimumSize">
                <size>
                 <width>2</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="0">
            <layout class="QFormLayout" name="formLayout_3">
             <property name="horizontalSpacing">
              <number>7</number>
             </property>
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>11</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>22</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_3">
               <property name="font">
                <font>
                 <family>Modern No. 20</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Phase</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="phase_shift">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="3" column="0">
            <layout class="QFormLayout" name="formLayout_5">
             <property name="horizontalSpacing">
              <number>7</number>
             </property>
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>11</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>22</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_5">
               <property name="font">
                <font>
                 <family>Modern No. 20</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Label</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="add_label">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <property name="toolTip">
        <string>Signal-to-Noise Ratio</string>
       </property>
       <attribute name="title">
        <string>Viewer</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_14">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox_3">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>MS Serif</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="title">
             <string>Original</string>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <property name="topMargin">
              <number>15</number>
             </property>
             <item row="0" column="0">
              <widget class="PlotWidget" name="widget3" native="true">
               <property name="minimumSize">
                <size>
                 <width>700</width>
                 <height>100</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QGroupBox" name="groupBox_4">
            <property name="font">
             <font>
              <family>MS Serif</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="title">
             <string>Reconsructed</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <property name="topMargin">
              <number>15</number>
             </property>
             <item row="0" column="0">
              <widget class="PlotWidget" name="widget4" native="true">
               <property name="minimumSize">
                <size>
                 <width>700</width>
                 <height>100</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QGroupBox" name="groupBox_5">
            <property name="font">
             <font>
              <family>MS Serif</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="title">
             <string>Error</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_7">
             <property name="topMargin">
              <number>16</number>
             </property>
             <item row="0" column="0">
              <widget class="PlotWidget" name="widget5" native="true">
               <property name="minimumSize">
                <size>
                 <width>700</width>
                 <height>100</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <widget class="QWidget" name="widget_2" native="true">
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>200</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>900</height>
           </size>
          </property>
          <layout class="QGridLayout" name="gridLayout_13">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_10">
             <item row="2" column="1">
              <widget class="QLCDNumber" name="snr_lcd">
               <property name="toolTip">
                <string>Signal-to-Noise Ratio</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <layout class="QFormLayout" name="formLayout_4">
               <item row="0" column="1">
                <widget class="QSlider" name="snr_slider">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="snr_label">
                 <property name="font">
                  <font>
                   <family>Modern No. 20</family>
                   <pointsize>10</pointsize>
                   <kerning>true</kerning>
                  </font>
                 </property>
                 <property name="text">
                  <string>SNR</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="11" column="0">
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="reset_button">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Reset</string>
               </property>
              </widget>
             </item>
             <item row="7" column="0" colspan="2">
              <widget class="QSlider" name="horizontalSlider_2">
               <property name="toolTip">
                <string>Sample Frequency Value</string>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="add_noise_button">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="toolTip">
                <string>Add Gaussian Noise to the loaded signal</string>
               </property>
               <property name="text">
                <string>Add Noise</string>
               </property>
              </widget>
             </item>
             <item row="9" column="1">
              <widget class="QLCDNumber" name="lcdNumber"/>
             </item>
             <item row="10" column="1">
              <widget class="QLCDNumber" name="lcdNumber_2">
               <property name="toolTip">
                <string>Error Value</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" rowspan="2" colspan="2">
              <widget class="QComboBox" name="freq_combobox">
               <property name="font">
                <font>
                 <family>MS Serif</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <item>
                <property name="text">
                 <string>select frequency</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Actual frequency</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Normal frequency</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="9" column="0">
              <widget class="QLabel" name="label_4">
               <property name="font">
                <font>
                 <family>Modern No. 20</family>
                 <pointsize>10</pointsize>
                 <weight>50</weight>
                 <bold>false</bold>
                </font>
               </property>
               <property name="text">
                <string>Slider_freq</string>
               </property>
              </widget>
             </item>
             <item row="10" column="0">
              <widget class="QLabel" name="label_6">
               <property name="font">
                <font>
                 <family>Modern No. 20</family>
                 <pointsize>10</pointsize>
                 <weight>50</weight>
                 <bold>false</bold>
                </font>
               </property>
               <property name="text">
                <string>Max_freq</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1350</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="load_option"/>
   </widget>
   <addaction name="menufile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="load_option">
   <property name="text">
    <string>Load Signal</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
